openedx/core/djangoapps/programs/tasks.py

```diff
@@ -53,14 +59,14 @@ def get_completed_programs(site, student):
         student (User): Representing the student whose completed programs to check for.
 
     Returns:
-        dict of {program_UUIDs: visible_dates}
+        set of program_UUIDs
 
     """
     meter = ProgramProgressMeter(site, student)
-    return meter.completed_programs_with_available_dates
+    return set(meter.completed_programs_with_available_dates.keys())
```

Comment:
after talking with @justinhynes, instead of modifying `completed_programs_with_available_dates`, I'm doing the safer thing and just taking the current return value and extracting out just the set of program UUIDs.  Re-factoring unnecessary code out of `ProgramProgressMeter` would be lovely but there are larger infrastructural improvements that would make that easier.

Reply:
I'm a lil worried about this change actually. I'm worried it might not work as intended. 

It looks like this piece of code is only used [right now in the `award_program_certificates` task](https://github.com/openedx/edx-platform/blob/e02a22f33847ba17db84cd7549fe4fc93c54f5fb/openedx/core/djangoapps/programs/tasks.py#L311-L314):
```
    try:
        completed_programs = {}
        for site in Site.objects.all():
            import pdb; pdb.set_trace()
            completed_programs.update(get_completed_programs(site, student))
```
We changed the return type of `get_completed_programs()` to a set, and `completed_programs` is a dict. I ran these changes locally in devstack after I got worried and I'm encountering a ValueError exception when we try to update the `completed_programs.update()` dict: `ValueError('dictionary update sequence element #0 has length 36; 2 is required')`